pymc/distributions/timeseries.py

```diff
@@ -179,122 +237,68 @@ def logp(self, value):
+
+    def logp(
+        value: at.Variable,
+        mu: at.Variable,
+        sigma: at.Variable,
+        init: at.Variable,
+    ) -> at.TensorVariable:
         """
         Calculate log-probability of Gaussian Random Walk distribution at specified value.
 
         Parameters
         ----------
-        x: numeric
-            Value for which log-probability is calculated.
+        value: at.Variable,
+        mu: at.Variable,
+        sigma: at.Variable,
+        init: at.Variable,
 
         Returns
         -------
         TensorVariable
         """
-        if x.ndim > 0:
-            x_im1 = x[:-1]
-            x_i = x[1:]
-            mu, sigma = self._mu_and_sigma(self.mu, self.sigma)
-            innov_like = Normal.dist(mu=x_im1 + mu, sigma=sigma).logp(x_i)
-            return self.init.logp(x[0]) + at.sum(innov_like)
-        return self.init.logp(x)
 
-    def random(self, point=None, size=None):
-        """Draw random values from GaussianRandomWalk.
+        # Calculate initialization logp
+        init_logp = pm.logp(Normal.dist(init, sigma), 0)
 
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
+        # Make time series stationary around the mean value
+        stationary_series = at.diff(value)
+        series_logp = pm.logp(Normal.dist(mu, sigma), stationary_series)
 
-        Returns
-        -------
-        array
-        """
-        # sigma, mu = distribution.draw_values([self.sigma, self.mu], point=point, size=size)
-        # return distribution.generate_samples(
-        #     self._random,
-        #     sigma=sigma,
-        #     mu=mu,
-        #     size=size,
-        #     dist_shape=self.shape,
-        #     not_broadcast_kwargs={"sample_shape": to_tuple(size)},
-        # )
-        pass
-
-    def _random(self, sigma, mu, size, sample_shape):
-        """Implement a Gaussian random walk as a cumulative sum of normals.
-        axis = len(size) - 1 denotes the axis along which cumulative sum would be calculated.
-        This might need to be corrected in future when issue #4010 is fixed.
-        """
-        if size[len(sample_shape)] == sample_shape:
-            axis = len(sample_shape)
-        else:
-            axis = len(size) - 1
-        rv = stats.norm(mu, sigma)
-        data = rv.rvs(size).cumsum(axis=axis)
-        data = np.array(data)
-
-        # the following lines center the random walk to start at the origin.
-        if len(data.shape) > 1:
-            for i in range(data.shape[0]):
-                data[i] = data[i] - data[i][0]
-        else:
-            data = data - data[0]
-        return data
+        total_logp = at.concatenate([at.expand_dims(init_logp, 0), series_logp])
```

Comment:
Used expands dims rather than `[:1]` to make the dimension expansion explicit rather than implicit